app/src/main/java/com/github/se/gomeet/ui/mainscreens/SearchModule.kt

```diff
@@ -0,0 +1,213 @@
+  val focusRequester = remember { FocusRequester() }
+  val keyboardController = LocalSoftwareKeyboardController.current
+
+  Column(modifier = Modifier.padding(16.dp)) {
+    TextField(
+        value = searchText,
+        leadingIcon = {
+          IconButton(onClick = { nav.navigateToScreen(Route.MESSAGE_CHANNELS) }) {
+            Icon(
+                ImageVector.vectorResource(R.drawable.gomeet_icon),
+                contentDescription = null,
+                tint = contentColor)
+          }
+        },
+        onValueChange = viewModel::onSearchTextChange,
+        modifier = Modifier.fillMaxWidth().focusRequester(focusRequester),
+        placeholder = { Text(text = "Search") },
+        keyboardOptions = KeyboardOptions.Default.copy(imeAction = ImeAction.Search),
+        keyboardActions =
+            KeyboardActions(
+                onSearch = {
+                  keyboardController?.hide()
+                  coroutineScope.launch { viewModel.performSearch(searchText) }
+                }))
+
+    Spacer(modifier = Modifier.height(16.dp))
+    if (isSearching) {
+      Box(modifier = Modifier.fillMaxSize()) {
+        CircularProgressIndicator(modifier = Modifier.align(Alignment.Center))
+      }
+    } else if (persons.isNotEmpty() && searchText.isNotEmpty()) {
+      LazyColumn(modifier = Modifier.fillMaxWidth().weight(1f)) {
+        items(persons) { item ->
+          SearchModuleSnippet(item, nav = nav, backgroundColor = backgroundColor)
+          Spacer(modifier = Modifier.height(8.dp))
+        }
+      }
+    }
+  }
+}
+
+@Composable
+fun SearchModuleSnippet(
+    item: SearchViewModel.SearchableItem,
+    nav: NavigationActions,
+    backgroundColor: Color
+) {
+  when (item) {
+    is SearchViewModel.SearchableItem.User -> {
+      val painter: Painter =
+          if (item.user.profilePicture.isNotEmpty()) {
+            rememberAsyncImagePainter(
+                ImageRequest.Builder(LocalContext.current)
+                    .data(data = item.user.profilePicture)
+                    .apply(
+                        block =
+                            fun ImageRequest.Builder.() {
+                              crossfade(true)
+                              placeholder(R.drawable.gomeet_icon)
+                            })
+                    .build())
+          } else {
+            painterResource(id = R.drawable.gomeet_icon)
+          }
+      Row(
+          verticalAlignment = Alignment.CenterVertically,
+          modifier =
+              Modifier.padding(vertical = 10.dp)
+                  .clickable {
+                    nav.navigateToScreen(Route.OTHERS_PROFILE.replace("{uid}", item.user.uid))
+                  }
+                  .background(color = backgroundColor, shape = RoundedCornerShape(10.dp))) {
+            Image(
+                painter = painter,
+                contentDescription = "User Icon",
+                modifier = Modifier.size(24.dp).padding(8.dp))
```

Comment:
Match profile picture style (see for example AddFriend screen) (round picture & bigger)